### ISSUE TITLE
`useOverlayTrigger` is missing `aria-haspopup` on most `triggerProps`

### ISSUE THREAD

**Issue by quisido** [adobe/react-spectrum#9542]
### Provide a general summary of the issue here

When using `{ type: 'dialog' }` or `grid` or `tree`, the resulting `triggerProps` does not contain `aria-haspopup="ROLE_HERE"`.

The `aria-haspopup` attribute only seems to be set when using `menu` or `listbox`.

### 🤔 Expected Behavior?

The `aria-haspopup` attribute on `triggerProps` should match the provided `type`.

### 😯 Current Behavior

The `aria-haspopup` attribute is missing for most types.

### 💁 Possible Solution

Add the `aria-haspopup` attribute for the remaining types.

### 🔦 Context

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-haspopup#dialog

### 🖥️ Steps to Reproduce

Relevant code snippet:
https://github.com/adobe/react-spectrum/blob/858d6cdd45bcf52ee33706a07f360bf68866a30b/packages/%40react-aria/overlays/src/useOverlayTrigger.ts#L53

>We only set it as menu because screen readers announce it as menu anyway.

This doesn't feel forward-supportive for screen readers that do add support for these other values. 🤔  Is there harm to setting it to `dialog` and having the screen reader announce `menu` anyway? At least it would add support for screen readers that _do_ support `dialog`.

### Version

N/A

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

MacOS

**Comment by snowystinger** [comment]
Thanks for the issue. We should probably check the support again, it's been >5 years since that comment was written. Would you like to have a start at gathering that info?

**Comment by quisido** [comment]
>Would you like to have a start at gathering that info?

@snowystinger I included the MDN article for `aria-haspopup` in the issue description. It lists the other roles as valid/expected values.

**Comment by snowystinger** [comment]
Thank you for that. Those all existed in MDN way back when as well, which is why that comment exists. The question is, are the announcements any different/better now. My guess as to why limited it, which isn't specifically said in the comment, is that in at least one of the combinations of browser+AT+OS, it did not work with anything other than 'menu' at that time. So yes, MDN says it should be supported, but I am wary that the support actually exists.
